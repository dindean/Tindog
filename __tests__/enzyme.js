import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import renderer from 'react-test-renderer';

import { addCard } from '../client/actions/actions';

// Newer Enzyme versions require an adapter to a particular version of React
configure({ adapter: new Adapter() });

describe('React unit tests', () => {
  describe('LabeledText', () => {
    let wrapper;
    const props = {
      label: 'Mega',
      text: 'Markets',
    };

    beforeAll(() => {
      wrapper = shallow(<LabeledText {...props} />);
    });

    it('Renders a <p> tag with the label in bold', () => {
      expect(wrapper.type()).toEqual('p');
      expect(wrapper.text()).toEqual('Mega: Markets');
      expect(wrapper.find('strong').text()).toMatch('Mega');
    });

    it('LabeledText renders correctly', () => {
      const tree = renderer
        .create(<LabeledText {...props} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('MarketDisplay', () => {
    let wrapper;
    const props = {
      percentage: 100,
      index: 2,
      location: 'Hogwarts',
      cards: 23,
      addCard: jest.fn(),
      deleteCard: jest.fn(),
    };

    beforeAll(() => {
      wrapper = shallow(<MarketDisplay {...props} />);
    });

    it('A MarketDisplay should display all of its text props inside a LabeledText component', () => {
      expect(wrapper.containsMatchingElement(<LabeledText label="Market ID" />)).toEqual(true);
      expect(wrapper.containsMatchingElement(<LabeledText label="Location" />)).toEqual(true);
      expect(wrapper.containsMatchingElement(<LabeledText label="Cards" />)).toEqual(true);
      expect(wrapper.containsMatchingElement(<LabeledText label="% of total" />)).toEqual(true);
    });

    it('It should also contain a div with two buttons', () => {
      expect(wrapper.find('div.flex').find('button').length).toEqual(2);
    });

    it('The functions passed down should be invoked on click', () => {
      props.addCard.mockClear();
      props.deleteCard.mockClear();
      wrapper.find('button').forEach((node) => {
        node.simulate('click');
      });
      expect(props.addCard).toHaveBeenCalled();
      expect(props.deleteCard).toHaveBeenCalled();
    });

    it('MarketDisplay should render a div with a class of `marketBox`, and the interior div wrapping the two buttons should have a class of `flex', () => {
      expect(wrapper.hasClass('marketBox')).toEqual(true);
      expect(wrapper.find('div.flex').length).toEqual(1);
    });

    it('MarketDisplay renders correctly', () => {
      const tree = renderer
        .create(<MarketDisplay {...props} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('MarketsDisplay', () => {
    let wrapper;
    const props = {
      totalCards: 15,
      marketList: [{ location: 'London', cards: 0 }, { location: 'Hogwarts', cards: 5 }, { location: 'Home', cards: 2 }],
      addCard: jest.fn(),
      deleteCard: jest.fn(),
    };

    beforeAll(() => {
      wrapper = shallow(<MarketsDisplay {...props} />);
    });

    it('A MarketsDisplay should have an h4 element to display the "Markets" title', () => {
      expect(wrapper.contains(<h4>Markets</h4>)).toEqual(true);
    });

    it('A single MarketDisplay is rendered for each market in the marketList prop', () => {
      // expect(wrapper.contains(<h4>Markets</h4>)).toEqual(true);
      expect(wrapper.find(MarketDisplay).length).toEqual(props.marketList.length);
    });

    // Test for zero, a whole number, and a fractional value. (Right now this
    // is implemented incorrectly, so follow TDD here)
    it('The percentage prop should be a string calculated to two decimals.', () => {
      wrapper.find(MarketDisplay).forEach(node => {
        expect(typeof node.props().percentage).toEqual('string');
        expect(node.props().percentage.split('.')[1].length).toEqual(2);
      });
    });

    it('MarketsDisplay renders correctly', () => {
      const tree = renderer
        .create(<MarketsDisplay {...props} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
