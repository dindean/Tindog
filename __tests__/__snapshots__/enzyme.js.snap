// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests LabeledText LabeledText renders correctly 1`] = `
<p>
  <strong>
    Mega: 
  </strong>
  Markets
</p>
`;

exports[`React unit tests MarketDisplay MarketDisplay renders correctly 1`] = `
<div
  className="marketBox"
>
  <p>
    <strong>
      Market ID: 
    </strong>
    2
  </p>
  <p>
    <strong>
      Location: 
    </strong>
    Hogwarts
  </p>
  <p>
    <strong>
      Cards: 
    </strong>
    23
  </p>
  <p>
    <strong>
      % of total: 
    </strong>
    100
  </p>
  <div
    className="flex"
  >
    <button
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
    >
      +
    </button>
    <button
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
    >
      -
    </button>
  </div>
</div>
`;

exports[`React unit tests MarketsDisplay MarketsDisplay renders correctly 1`] = `
<div
  className="displayBox"
>
  <h4>
    Markets
  </h4>
  <div
    className="allMarkets"
  >
    <div
      className="marketBox"
    >
      <p>
        <strong>
          Market ID: 
        </strong>
        0
      </p>
      <p>
        <strong>
          Location: 
        </strong>
        London
      </p>
      <p>
        <strong>
          Cards: 
        </strong>
        0
      </p>
      <p>
        <strong>
          % of total: 
        </strong>
        0.00
      </p>
      <div
        className="flex"
      >
        <button
          onClick={[Function]}
        >
          +
        </button>
        <button
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </div>
    <div
      className="marketBox"
    >
      <p>
        <strong>
          Market ID: 
        </strong>
        1
      </p>
      <p>
        <strong>
          Location: 
        </strong>
        Hogwarts
      </p>
      <p>
        <strong>
          Cards: 
        </strong>
        5
      </p>
      <p>
        <strong>
          % of total: 
        </strong>
        33.33
      </p>
      <div
        className="flex"
      >
        <button
          onClick={[Function]}
        >
          +
        </button>
        <button
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </div>
    <div
      className="marketBox"
    >
      <p>
        <strong>
          Market ID: 
        </strong>
        2
      </p>
      <p>
        <strong>
          Location: 
        </strong>
        Home
      </p>
      <p>
        <strong>
          Cards: 
        </strong>
        2
      </p>
      <p>
        <strong>
          % of total: 
        </strong>
        13.33
      </p>
      <div
        className="flex"
      >
        <button
          onClick={[Function]}
        >
          +
        </button>
        <button
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </div>
  </div>
</div>
`;
